### PR TITLE
Dockerfile: reduce layer bloat

### PR DESCRIPTION
**Pull request by DrMurx**
### Background

The current Dockerfile generates a large amount of unneeded layers containing copies of the APT cache at build time.  

### Changes

A GPT model advised to modify the Dockerfile like this. ;-)

### Documentation

This is what the GPT model had to say about its changes:

In this new version, we've merged all installation steps in a single `RUN` command. This way, we can reduce the number of layers in the image and avoid bloating it unnecessarily. We first install all the dependencies except Chromium and Firefox-ESR. After that, we add the Google repository key and repository to the sources list, and then we update the package lists again and install Chromium and Firefox-ESR. We've added a clean step to remove the apt cache and other unused files, reducing the size of the resulting layer.

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [ ] n/a I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [ ] n/a I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Review by hdkiller — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1,19 +1,15 @@
...
-# Install Xvfb and other dependencies for headless browser testing
-RUN apt-get update \
-    && apt-get install -y wget gnupg2 libgtk-3-0 libdbus-glib-1-2 dbus-x11 xvfb ca-certificates
-
-# Install Firefox / Chromium
-RUN wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \
+# Install dependencies
+RUN apt-get -y update \
+    && apt-get -y install git chromium-driver wget gnupg2 libgtk-3-0 libdbus-glib-1-2 dbus-x11 xvfb ca-certificates \
+    && wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \
```
Comment (Wladastic) [inline Dockerfile]: I know adding && makes it more safe but it's super hard to read now

Comment (DrMurx) [reply Dockerfile]: That's more a matter of habits. If I write a Dockerfile occasionally, I might not be used to that syntax. If I do it every day, I don't even notice the `&&` and the `\` anymore.

Btw, it has nothing to do with "more safety", as there's no "less safe" alternative to it. It's a hard requirement that all commands in each `RUN` line get executed, and if only one of them fails, the docker build should fail too.

Creating docker images without apt cache clutter is simply best practice. Here's what a GPT model has to say about this:

> When creating a Dockerfile, it's generally a good practice to execute `apt-get update` to ensure that the package list is up to date before installing any packages. However, it's also a good practice to clean up the cache in the same layer to keep the image size small.

**Comment by richbeales** [comment]
Anecdotally our Docker Build CI task took twice as long on this PR.  Would want to check that before committing.

**Comment by DrMurx** [comment]
> Anecdotally our Docker Build CI task took twice as long on this PR. Would want to check that before committing.

@richbeales That may very well be and it heavily depends on how you've organized your builds. 

Fewer container image layers also means fewer opportunities for the [build cache](https://docs.docker.com/build/cache/). I would argue that layer caching is most beneficial during development where you execute `docker build` frequently in your local environment, but a CI build job should always start clean (ie `docker build --no-cache`) to avoid stale dependencies.

**Review by ntindle — APPROVED** [review]

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by Boostrix** [comment]
would also be interested in less bloat :+1:

**Review by Pwuts — CHANGES_REQUESTED** [review]
Cleaning up the apt cache is good, but please maintain the layer separation we have now for caching sake.

**Review by Pwuts — APPROVED** [review]

**Comment by Pwuts** [comment]
This is saving ~18MB overall.. just over 1% of the release image :')

**Pull request merged**